From 207071307ff78cddabc0d88da6f67c26dc1e6166 Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Wed, 5 Sep 2018 11:30:39 +0200
Subject: [PATCH 0694/1051] arm64: dts: marvell: armada-37xx: Add comphy node

This patch adds a comphy node describing the marvell common phy found in
Armada 37xx SoCs.

Change-Id: I4807f029807620427622781af3e0259374eb3742
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-37xx.dtsi | 25 ++++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
index 35aa1bc85e12..22ce7f107019 100644
--- a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
@@ -164,6 +164,31 @@
 				#clock-cells = <1>;
 			};
 
+			a3700_comphy: comphy@18300 {
+				compatible = "marvell,comphy-a3700";
+				reg = <0x18300 0x300>, <0x1f000 0x1000>,
+				      <0x5c000 0x1000>, <0xe0178 0x8>;
+				reg-names = "comphy", "serdes",
+					    "usb3_gbe1_phy", "indirect_ahci";
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				a3700_comphy0: phy@0 {
+					reg = <0>;
+					#phy-cells = <1>;
+				};
+
+				a3700_comphy1: phy@1 {
+					reg = <1>;
+					#phy-cells = <1>;
+				};
+
+				a3700_comphy2: phy@2 {
+					reg = <2>;
+					#phy-cells = <1>;
+				};
+			};
+
 			tbg: tbg@13200 {
 				compatible = "marvell,armada-3700-tbg-clock";
 				reg = <0x13200 0x100>;
-- 
2.17.1

